packages/shared/src/loadClerkJsScript.ts

```diff
@@ -0,0 +1,128 @@
+import type { ClerkOptions, SDKMetadata, Without } from '@clerk/types';
+
+import { createDevOrStagingUrlCache, parsePublishableKey } from './keys';
+import { loadScript } from './loadScript';
+import { isValidProxyUrl, proxyUrlToAbsoluteURL } from './proxy';
+import { addClerkPrefix } from './url';
+import { versionSelector } from './versionSelector';
+
+const FAILED_TO_LOAD_ERROR = 'Clerk: Failed to load Clerk';
+const MISSING_PUBLISHABLE_KEY_ERROR =
+  'Clerk: Missing publishableKey. You can get your key at https://dashboard.clerk.com/last-active?path=api-keys.';
+
+const { isDevOrStagingUrl } = createDevOrStagingUrlCache();
+
+type LoadClerkJsScriptOptions = Without<ClerkOptions, 'isSatellite'> & {
+  publishableKey: string;
+  clerkJSUrl?: string;
+  clerkJSVariant?: 'headless' | '';
+  clerkJSVersion?: string;
+  sdkMetadata?: SDKMetadata;
+  proxyUrl?: string;
+  domain?: string;
+  /**
+   * @internal
+   * The version of `@clerk/clerk-js` that will be used if an explicit version
+   * is not provided. For `@clerk/clerk-react`, this will be the PACKAGE_VERSION.
+   */
+  __internal_packageVersion?: string;
```

Comment:
This additional property is used internally to provide a specific package version that will be passed through to the clerk-js script load.

The reason we're adding this is to keep the current behavior of the version selector used by [`@clerk/clerk-react`](https://github.com/clerk/javascript/blob/d8d752c527d8d586bf730991c4c33930bbf743e1/packages/react/src/utils/versionSelector.ts#L11), where the `packageVersion` is equal to the `@clerk/clerk-react` version. For other SDKs, this will default to the `@clerk/clerk-js` version.

More info [here](https://github.com/clerk/javascript/pull/3681#discussion_r1673544525).

Reply:
I believe we can drop the usage of `PACKAGE_VERSION` from versionSelector of `clerk-react` because for that package `PACKAGE_VERSION` and `JS_PACKAGE_VERSION` majors always match

Reply:
Answering to [this](https://github.com/clerk/javascript/pull/3849#issuecomment-2269259335). If in the future we want to have a different majors between clerk-react and clerk-js, we can update the versionSelector then. Even with the current version if `__internal_packageVersion` was a different major from the `clerk-js` one, without any other changes, that setup would be broken because clerk-react would attempt to load v5 where a new clerk-js v6 would exist.

Reply:
Thanks for explaining @panteliselef. That makes sense then. For us to move forward we can set the default `packageVersion` to `JS_PACKAGE_VERSION` since they will have the same major version anyway:

https://github.com/clerk/javascript/blob/7e94fcf0fcbee8842a54f7931c45190370aa870d/packages/react/src/utils/versionSelector.ts#L34

But I'm not sure what will happen to prerelease tags, particularly on line 22 here

https://github.com/clerk/javascript/blob/7e94fcf0fcbee8842a54f7931c45190370aa870d/packages/react/src/utils/versionSelector.ts#L16-L26